grpc_server.go

```diff
@@ -196,16 +197,19 @@ func filterMetadata(hw []byte, filter string) ([]byte, error) {
 		if err, ok := v.(error); ok {
 			return nil, err
 		}
-		result = v
-	}
 
-	if resultString, ok := result.(string); ok { // if already a string, don't marshal
-		return []byte(resultString), nil
-	}
-	if result != nil { // if nil, don't marshal (json.Marshal(nil) returns "null")
-		return json.Marshal(result)
+		if vString, ok := v.(string); ok { // if already a string, don't marshal
+			result = fmt.Sprintln(result + vString) // append v to result
```

Comment:
this doesn't really make sense, since first you are concatenating both strings and then basically just returning it but with an extra new-line (from the `ln` family) which then gets stripped off when returning at the end.

These calls can all just be `result += vString` or `result += string(vByte)` followed with just `return []byte(result)`

Reply:
The new line is really for the list values returned as separate results (i.e. tags and ssh keys). If I don't add a new line at the end of each iteration, `["tag1", "tag2"]` will become `tag1tag2` as the result, instead of 
```
tag1
tag2
```
I remove the new line at the very end (outside of the loop) so it matches the behavior in Kant, which also doesn't return a new line at the end of the response.

For _some reason_, I thought `+=` was not a thing in golang.... I'll reformat that..

Reply:
Oh, instead of using fmt.Sprint? Is there a reason why?